crypto/engine/eng_linux_getrandom.c

```diff
@@ -0,0 +1,150 @@
+#include <openssl/opensslconf.h>
+
```

Comment:
Not that my opinion matters all that much, but yes. Adding `getrandom()` through a new engine means that only the people who enable the engine are going to benefit from this, but it would also bypass `md_rand` impacting performance significantly.

On the other hand adding this to `RAND_poll()` means that `getrandom()` is only used to seed the user-space PRNG and that we don't need a syscall every time random data is needed (the RNG on Linux is also quite susceptible to lock contention).

(The fact that `md_rand` is not exactly optimal and needs to be rewritten is a different problem).